datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -506,6 +506,71 @@ mod tests {
         assert_batches_eq!(exp, collected.as_slice());
     }
 
+    #[tokio::test]
+    async fn test_merge_five_partitions() {
```

Comment:
I wonder if you can leave some comments about why this test was added / what it is covering to help future readers?

I think the coverage of merging is pretty good with this: https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/merge_fuzz.rs

Reply:
Ah yes, I added that test just to convince myself that the implementation of loser tree works.

Now I'm pretty sure it works, perhaps we can remove it?

Reply:
I leave that to your judgement